### PR TITLE
[vs][multi-asic]: Add support to build multi-asic vs images

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
#### Why I did it
Pre-requisite: https://github.com/Azure/sonic-buildimage/pull/8269
- To be able to generate multi-asic KVM image.
- To provide flexibility to generate single asic or both single and multi-asic images.
#### How I did it
- Add a new build param, BUILD_MULTIASIC_KVM, if set to "y", the multi-asic VS target KVM images will be generated. If not, only single asic VS image will be generated.
- Make changes to build_image.sh to generate 4-asic and 6-asic KVM images if BUILD_MULTIASIC_KVM parameter is set to y
#### How to verify it
Generate single-asic VS as currently done, no change in build steps:
make configure PLATFORM=vs
make target/sonic-vs.img.gz - will generate only single asic KVM image.

make BUILD_MULTIASIC_KVM=y target/sonic-vs.img.gz - will generate single asic and multi-asic KVM images.
should generate:
sonic-vs.bin
sonic-vs.img.gz
sonic-4asic-vs.img.gz
sonic-6asic-vs.img.gz 
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: onie-image.conf [hunk 1]
```diff
@@ -50,3 +50,10 @@ OUTPUT_ABOOT_IMAGE=target/sonic-aboot-$TARGET_MACHINE.swi
 
 ## Aboot boot image name
 ABOOT_BOOT_IMAGE=.sonic-boot.swi
+
+## Output file name for 4-asic mkvm image
+OUTPUT_MKVM_4ASIC_IMAGE=target/sonic-4asic-$TARGET_MACHINE.img
```
Comment (judyjoseph) [inline onie-image.conf:55]: We could update the other files for other arch's as well, 
onie-image-arm64.conf
onie-image-armhf.conf

File: build_image.sh [hunk 2]
```diff
@@ -25,6 +25,46 @@ fi
 
 IMAGE_VERSION="${SONIC_IMAGE_VERSION}"
 
+generate_kvm_image()
+{
+    NUM_ASIC=$1
+    if [ $NUM_ASIC == 1 ]; then
```
Comment (abdosi) [inline build_image.sh:31]: make default as 1

File: build_image.sh [hunk 3]
```diff
@@ -93,32 +133,16 @@ elif [ "$IMAGE_TYPE" = "raw" ]; then
...
-
     generate_onie_installer_image
-
-    SONIC_USERNAME=$USERNAME PASSWD=$PASSWORD sudo -E ./scripts/build_kvm_image.sh $KVM_IMAGE_DISK $onie_recovery_image $OUTPUT_ONIE_IMAGE $KVM_IMAGE_DISK_SIZE
-
-    if [ $? -ne 0 ]; then
-        echo "Error : build kvm image failed"
-        exit 1
+    # Generate single asic KVM image
+    generate_kvm_image 1
```
Comment (abdosi) [inline build_image.sh:138]: no need to pass 1

File: build_image.sh [hunk 4]
```diff
@@ -25,6 +25,46 @@ fi
...
+         RECOVERY_ISO=$onie_recovery_image
+    elif [ $NUM_ASIC == 4 ]; then 
+         KVM_IMAGE=$OUTPUT_MKVM_4ASIC_IMAGE
+         RECOVERY_ISO=$onie_recovery_mkvm_4asic_image
+    elif [ $NUM_ASIC == 6 ]; then
+         KVM_IMAGE=$OUTPUT_MKVM_6ASIC_IMAGE
+         RECOVERY_ISO=$onie_recovery_mkvm_6asic_image
+    fi
+
+    echo "Build KVM image"
```
Comment (abdosi) [inline build_image.sh:42]: update  log message to include num asic

File: onie-image.conf (hunk 1 above)
Comment (abdosi) [reply onie-image.conf:55]: we can add this later i think.. as we can not verify these environment

File: rules/config [hunk 5]
```diff
@@ -196,3 +196,6 @@ SONIC_VERSION_CONTROL_COMPONENTS ?= none
 # Set the env variable ENABLE_DOCKER_BASE_PULL = y to enable pulling sonic-slave docker from registry
 REGISTRY_PORT ?= 443
 REGISTRY_SERVER ?= sonicdev-microsoft.azurecr.io
+
+# Default number of asics in VS platform
```
Comment (abdosi) [inline rules/config:200]: fix the comment

File: build_image.sh (hunk 2 above)
Comment (SuvarnaMeenakshi) [reply build_image.sh]: Updated as per comment.

File: build_image.sh (hunk 3 above)
Comment (SuvarnaMeenakshi) [reply build_image.sh]: Updated as per comment.

File: build_image.sh (hunk 4 above)
Comment (SuvarnaMeenakshi) [reply build_image.sh]: Updated as suggested.

File: rules/config (hunk 5 above)
Comment (SuvarnaMeenakshi) [reply rules/config]: Updated comment.

**Review by abdosi — DISMISSED** [review]
LGTM

**Review by judyjoseph — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]

**Pull request merged**